src/main/java/nextstep/subway/section/domain/Sections.java

```diff
@@ -0,0 +1,111 @@
+                .line(line)
+                .upStation(upStation)
+                .downStation(downStation)
+                .distance(distance)
+                .build());
+    }
+
+    public List<Station> getStations() {
+        List<Station> stations = new ArrayList<>();
+        Section upEndSection = findUpEndSection();
+        stations.add(upEndSection.getUpStation());
+        Section nextSection = upEndSection;
+        while (nextSection != null) {
+            stations.add(nextSection.getDownStation());
+            nextSection = findSectionByNextUpStation(nextSection.getDownStation());
+        }
+        return stations;
+    }
+
+    private Section findUpEndSection() {
+        List<Station> downStations = this.sections.stream()
+                .map(Section::getDownStation)
+                .collect(toList());
+        return this.sections.stream()
+                .filter(section -> !downStations.contains(section.getUpStation()))
+                .findFirst()
+                .orElseThrow(EntityNotFoundException::new);
+    }
+
+    private Section findSectionByNextUpStation(Station station) {
+        return this.sections.stream()
+                .filter(section -> section.getUpStation() == station)
+                .findFirst()
+                .orElse(null);
+    }
+
+    public void add(Section section) {
+        validateNotExistStation(section);
+        alreadyExistStation(section);
+
+        addUpStation(section);
+        addDownStation(section);
+
+        this.sections.add(section);
+    }
+
+    private void addDownStation(Section section) {
+        sections.stream()
+                .filter(oldSection -> section.getDownStation() == oldSection.getDownStation())
+                .findFirst()
+                .ifPresent(oldSection -> {
+                    selectWithStation(section, oldSection, oldSection.getUpStation(), section.getUpStation());
+                });
+    }
+
+    private void addUpStation(Section section) {
+        sections.stream()
+                .filter(oldSection -> section.getUpStation() == oldSection.getUpStation())
+                .findFirst()
+                .ifPresent(oldSection -> {
+                    selectWithStation(section, oldSection, section.getDownStation(), oldSection.getDownStation());
+                });
+    }
+
+    private void selectWithStation(Section section, Section oldSection, Station upStation, Station upStation2) {
+        if (oldSection.getDistance() <= section.getDistance()) {
+            CommonException.IllegalArgumentException(Message.DISTANCE_EXCESS_MESSAGE);
+        }
+        sections.add(new Section(oldSection.getLine(), upStation, upStation2, oldSection.getDistance() - section.getDistance()));
+        sections.remove(oldSection);
+    }
+
+    private void validateNotExistStation(Section section) {
+        if(!getStations().contains(section.getUpStation()) && !getStations().contains(section.getDownStation())) {
```

Comment:
하나의 리스트로 재사용할 수 있지 않을까요?

```suggestion
        List<Station> stations = getStations();
        if(!stations.contains(section.getUpStation()) && !stations.contains(section.getDownStation())) {
```

Reply:
지하철역 조회 부분을 불필요하게 2번 호출하여 검증을 했네요
좋은 피드백 감사 드립니다. 🙇‍♂️